src/game.py

```diff
@@ -88,119 +85,129 @@ def duel(
+                    print_statements.append(f"Round {round}, Duel {i+1}")
+                    print_statements.append(
+                        "=================================================="
                     )
                 winner_key = self.duel(duel_pairs, self.spells)
-                if (
-                    i + 1 == len(character_pairs_list)
-                    and round != self.total_rounds
-                ):
-                    print(
-                        "==================================================\n"
+                winner_index = winner_key[0]
+                print_statements.append(winner_key[-1])
+                if i + 1 == len(character_pairs_list) and round != self.total_rounds:
+                    print_statements.append(
+                        "=================================================="
                     )
-                    print(f"Round {round} is over!\n")
+                    print_statements.append(f"Round {round} is over!")
                 else:
                     pass
-                temp_winners.append(duel_pairs[winner_key])
+                temp_winners.append(duel_pairs[winner_index])
             if len(temp_winners) == 1:
-                self.print_winner(temp_winners[0])
-                if temp_winners[0]["Name"] == self.champion:
-                    print(f"Your champion is victorious!")
+                winner_name = temp_winners[0]["Name"]
+                print_statements.append(self.print_winner(temp_winners[0]))
+                """if temp_winners[0]["Name"] == self.champion:
+                    print_statements.append(f"Your champion is victorious!")
                 else:
-                    print(
+                    print_statements.append(
                         f"Your champion {self.champion} did not make it "
-                        f"to the end...\n"
-                    )
+                        f"to the end..."
+                    )"""
                 break
             else:
                 self.characters_playing = temp_winners
                 character_pairs_list = self.bracket_maker()
                 temp_winners = []
                 round += 1
                 if len(character_pairs_list) == 1:
-                    print("==================================================")
-                    print("Final Duel!\n")
-                    print(
-                        "Duel: ",
-                        duel_pairs[0]["Name"],
-                        "vs.",
-                        duel_pairs[1]["Name"],
+                    print_statements.append(
+                        "=================================================="
                     )
-                    print("   ")
+                    print_statements.append("Final Duel!")
+                    print_statements.append(
+                        f"{duel_pairs[0]['Name']} vs. {duel_pairs[1]['Name']}"
```

Comment:
hi rachel, to address the wrong `Duel: ` in the final duel, here's my suggestion, and it's tested on my local machine.
```suggestion
                        f"{self.characters_playing[0]['Name']} vs. {self.characters_playing[1]['Name']}"
```
I mistakenly take a previous tuple containing the second-to-last fighters as the final fighters. This change fixed it.